keras/legacy/interfaces.py

```diff
@@ -141,3 +141,54 @@ def lstm_args_preprocessor(args, kwargs):
     value_conversions={'dim_ordering': {'tf': 'channels_last',
                                         'th': 'channels_first',
                                         'default': None}})
+
+legacy_evaluate_generator_support = generate_legacy_interface(
```

Comment:
The error messages in `generate_legacy_interface` are layer-specific. Don't use it for model methods.